### PR TITLE
delayresume: Add new plugin

### PR DESCRIPTION
**Pull request by rohgarg**
The README file describes the plugin, usage, and use-case.

### CODE REVIEW

File: contrib/delayresume/delayresume.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+dmtcp_checkpoint_block_resume()
+{
+  int ret = DMTCP_NOT_PRESENT;
+  ckptInitiator = true;
+  blockUserThreads();
+  ret = dmtcp_checkpoint();
+  if (ret == DMTCP_NOT_PRESENT) {
+    unblockUserThreads();
+    ckptInitiator = false;
+  } else if (ret == DMTCP_AFTER_RESTART) {
```
Comment (jiajuncao) [inline contrib/delayresume/delayresume.cpp]: The two branches do the same thing.

**Comment by jiajuncao** [comment]
Looks good.

I have one question regarding the motivation: if the block/unblock is done after resuming, all checkpoint-related events have finished, how does user's behavior at this point affect the checkpoint performance?

File: contrib/delayresume/delayresume.cpp (hunk 1 above)
Comment (rohgarg) [inline contrib/delayresume/delayresume.cpp]: Good point! I'll rework this.

**Comment by rohgarg** [comment]
I'm not sure what you mean by the checkpoint performance here. The goal is to allow a user thread that requested for a checkpoint to block other user threads from resuming until the checkpoint-initiator gives a go-ahead.

**Pull request merged**